kymatio/scattering2d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,181 @@
+            `1` or `2`. Defaults to `2`.
+        pre_pad : boolean, optional
+            Controls the padding: if set to False, a symmetric padding is applied
+            on the signal. If set to true, the software will assume the signal was
+            padded externally. Defaults to `False`.
+
+        Attributes
+        ----------
+        J : int
+            Log-2 of the scattering scale.
+        shape : tuple of int
+            Spatial support (M, N) of the input.
+        L : int, optional
+            Number of angles used for the wavelet transform.
+        max_order : int, optional
+            The maximum order of scattering coefficients to compute.
+            Must be either equal to `1` or `2`.
+        pre_pad : boolean
+            Controls the padding: if set to False, a symmetric padding is applied
+            on the signal. If set to true, the software will assume the signal was
+            padded externally.
+        Psi : dictionary
+            Contains the wavelets filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        Phi : dictionary
+            Contains the low-pass filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        M_padded, N_padded : int
+             Spatial support of the padded input.
+
+        Notes
+        -----
+        The design of the filters is optimized for the value L = 8.
+
+        pre_pad is particularly useful when cropping bigger images because
+        this does not introduce border effects inherent to padding.
+
+        """
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False):
+        super(Scattering2D, self).__init__(J, shape, max_order=max_order)
+        self.pre_pad, self.L = pre_pad, L
+        if 2 ** J > shape[0] or 2 ** J > shape[1]:
+            raise RuntimeError('The smallest dimension should be larger than 2^J')
+        self.build()
+
+    def build(self):
+        self.M, self.N = self.shape
+        self.modulus = Modulus()
+        self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
+        # pads equally on a given side if the amount of padding to add is an even number of pixels, otherwise it adds an extra pixel
+        self.pad = Pad([(self.N_padded - self.N) // 2, (self.N_padded - self.N + 1) // 2, (self.M_padded - self.M) // 2,
+                       (self.M_padded - self.M + 1) // 2], [self.N, self.M], pre_pad=self.pre_pad)
+        self.subsample_fourier = SubsampleFourier()
+        self.create_and_register_filters()
+
+    def create_and_register_filters(self):
+        """ This function run the filterbank function that
+            will create the filters as numpy array, and then, it
+            saves those arrays as module's buffers."""
+
+        # Create the filters
+        filters = filter_bank(self.M_padded, self.N_padded, self.J, self.L)
+        n = 0
+        self.phi, self.psi = filters['phi'], filters['psi']
```

Comment:
cosmetic: These should probably not be stored in `self` here, but rather just in names `phi`, `psi`, and then stored in a new dictionary lower down. probably don't change, since it might require adding boiler plate code

Reply:
well, for now, I view it more as hack: for instance, it allows to (possibly) memorize in which order the buffers were stored